crates/ty_python_semantic/src/symbol.rs

```diff
@@ -318,16 +318,28 @@ pub(crate) fn imported_symbol<'db>(
     // ignore `__getattr__`. Typeshed has a fake `__getattr__` on `types.ModuleType` to help out with
     // dynamic imports; we shouldn't use it for `ModuleLiteral` types where we know exactly which
     // module we're dealing with.
-    symbol_impl(db, global_scope(db, file), name, requires_explicit_reexport).or_fall_back_to(
-        db,
-        || {
-            if name == "__getattr__" {
-                Symbol::Unbound.into()
-            } else {
-                KnownClass::ModuleType.to_instance(db).member(db, name)
-            }
-        },
-    )
+    let module_type_member = || {
+        if name == "__getattr__" {
+            Symbol::Unbound.into()
+        } else {
+            KnownClass::ModuleType.to_instance(db).member(db, name)
+        }
+    };
+
+    let result = symbol_impl(db, global_scope(db, file), name, requires_explicit_reexport)
+        .or_fall_back_to(db, module_type_member);
+
+    if result
+        .symbol
+        .ignore_possibly_unbound()
+        .is_some_and(|ty| ty.is_never())
```

Comment:
One rather obvious problem here is a module that actually exports a symbol with type Never. I need to think about this..

Reply:
This is now fixed.